backend/infrahub/core/relationship/model.py

```diff
@@ -1148,3 +1154,15 @@ async def to_graphql(
             return None
 
         return await relationships[0].to_graphql(fields=fields, db=db, related_node_ids=related_node_ids)
+
+    async def _validate_hierarchy(self) -> None:
+        if self.node._schema.is_profile_schema or not self.node._schema.hierarchy:  # type: ignore[union-attr]
```

Comment:
we should use `node.get_schema` in this method instead of accessing the `_schema` private variable